applications/nrf_desktop/doc/dvfs.rst

```diff
@@ -0,0 +1,120 @@
+.. _nrf_desktop_dvfs:
+
+DVFS module
+###########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The nRF Desktop Dynamic Voltage and Frequency Scaling (DVFS) module is responsible for switching the frequency and voltage according to the application needs.
+The module chooses the highest frequency required by the application states.
+The module, requests this frequency as a minimal requirement to clock control driver which selects DVFS operating point accordingly.
+DVFS improves balancing performance and power consumption, especially in battery-operated devices.
+
+In the nRF Desktop application, the DVFS module enables the system to use the lowest possible frequency and voltage when the application performs less compute-intensive activities, such as Bluetooth® LE advertising.
+However, if performing more compute-intensive tasks, the nRF Desktop DVFS module will dynamically change to a higher frequency to increase performance.
+
+The nRF Desktop DVFS module requests clock frequency change using API of the clock control driver.
+The driver is responsible for choosing the optimal clock frequency.
+The driver submits a request to the sysctrl core through the nrfs DVFS service.
+The sysctrl core controls both clock frequency and voltage and chooses corresponding voltage to requested frequency.
+
+.. note::
+   In the current solution, it is assumed that no other module requests frequency changes.
+   The nRF Desktop DVFS module is the sole entity responsible for managing frequency and voltage adjustments.
+   Controlling clock from other sources would result in a module fatal error.
+
+Supported devices
+*****************
+
+The module is only available for the nRF54H20 SoC.
+
+The nRF54H20 SoC has the following available operating points (called frequency in the documentation):
+
+* DVFS_FREQ_HIGH - High frequency - 320 MHz, 0.8 V
+* DVFS_FREQ_MEDLOW - Medium-low frequency - 128 MHz, 0.6V
+* DVFS_FREQ_LOW - Low frequency - 64 MHz, 0.5V
+
+The nRF Desktop DVFS module might request any clock frequency, but it would be rounded up by the driver to one of the values from available operating points.
+
+Module events
+*************
+
+.. include:: event_propagation.rst
+    :start-after: table_dvfs_start
+    :end-before: table_dvfs_end
+
+.. note::
+    |nrf_desktop_module_event_note|
+
+Configuration
+*************
+
+The nRF Desktop DVFS module is enabled by default on supported board targets.
+To disable it, set the :option:`CONFIG_DESKTOP_DVFS` Kconfig option to ``n``.
+
+Tracked application states
+==========================
+
+The nRF Desktop DVFS module defines independent application states that have different requirements for each DVFS operating point.
```

Comment:
`have different requirements in terms of DVFS operating point`? `that require using different DVFS operating point`?

Reply:
Also I would use term `frequency` instead of `DVFS operating point` (we have a paragraph that explains how `frequency` maps to `DVFS operating point` and what happens under the hood - it should be enough). Still, I would not insist